src/main/java/seedu/exercise/ui/BarChartPanel.java

```diff
@@ -0,0 +1,59 @@
+package seedu.exercise.ui;
+
+import java.util.ArrayList;
+
+import javafx.fxml.FXML;
+import javafx.scene.chart.BarChart;
+import javafx.scene.chart.CategoryAxis;
+import javafx.scene.chart.NumberAxis;
+import javafx.scene.chart.XYChart;
+import javafx.scene.layout.Region;
+import seedu.exercise.logic.commands.statistic.Statistic;
+
+/**
+ * An UI for bar chart.
+ */
+public class BarChartPanel extends UiPart<Region> {
+    private static final String FXML = "BarChartPanel.fxml";
+
+    private Statistic statistic;
+
+    @FXML
+    private CategoryAxis xAxis;
+    @FXML
+    private NumberAxis yAxis;
+    @FXML
+    private BarChart<String, Double> barChart;
+
+    public BarChartPanel(Statistic statistic) {
+        super(FXML);
+        this.statistic = statistic;
+        display();
+    }
+
+    /**
+     * Set data for bar chart to be displayed.
+     */
+    private void display() {
+        barChart.setAnimated(false);
+        barChart.layout();
+
+        xAxis.setLabel(statistic.getCategory());
+        yAxis.setLabel("quantity");
+
+        XYChart.Series series = new XYChart.Series();
+
+        ArrayList<String> names = statistic.getProperties();
+        ArrayList<Double> values = statistic.getValues();
+
+        int size = names.size();
+        for (int i = 0; i < size; i++) {
+            series.getData().add(new XYChart.Data<>(names.get(i), values.get(i)));
+        }
+
+        barChart.setLegendVisible(false);
+        barChart.setTitle(statistic.getCategory() + " (" + statistic.getStartDate()
+                + " to " + statistic.getEndDate() + ")");
```

Comment:
May want to abstract this portion out into a `Util` class so all your `xxxPanels` can use it to format the title text.

Reply:
done